POI bubble start

ErikaJZhou opened the pull request:
Basically creates a plane behind the test text that could be used as the backing for the POI info bubble. 
This isn't a finished bubble yet. It still needs to have the business info and rating added. Eventually, the info needs to be pulled from the API.
We may need to coordinate some frontend code organization and it seems Jules and Justin made some changes as well.
![grey-poi-bubble-start-10-15-23](https://github.com/Augmenters/FrontStone/assets/55419983/10f1cf5e-e188-41b2-824b-ee95b7925858)

ErikaJZhou committed:
add app icon to About View

ErikaJZhou committed:
try to add blue plane

ErikaJZhou committed:
large blue plane shows up and also acts clickable

ErikaJZhou committed:
move file

ErikaJZhou committed:
try fix conflict

jm564 committed:
Add slider and crime stuff

Mostly done, just waiting on the crime table to be populated in the database to test with real data

ErikaJZhou committed:
Merge branch 'poi-bubble' of github.com:Augmenters/FrontStone into poi-bubble

ErikaJZhou committed:
add files to Git

Tap and CityARView which might have been removed by refactor

ErikaJZhou committed:
Merge remote-tracking branch 'origin/main' into poi-bubble

ErikaJZhou committed:
add file for deletion

jm564 committed:
Reorganize things so that we can use the AR view

jm564 committed:
Create Package.resolved

jm564 committed:
Merge branch 'add_shaded_crime_regions' into refactor

ErikaJZhou committed:
fix typo in camera permissions dialog

ErikaJZhou committed:
add blue plane test back

ErikaJZhou committed:
smaller plane panel

ErikaJZhou committed:
changed bubble color to grey

ErikaJZhou committed:
moved plane creation code to function

ErikaJZhou committed:
test positioning 3D text using containerFrame

ErikaJZhou committed:
poi bubble text uses POI struct

ErikaJZhou committed:
function for placing text at specific offset

ErikaJZhou committed:
font size parameter

ErikaJZhou committed:
rough poi bubble layout with mock Harpos data

ErikaJZhou commented:
![poi-bubble-harpos-test](https://github.com/Augmenters/FrontStone/assets/55419983/9c46a835-9a94-498a-9874-7c2fe5eb00a3)
POI bubble test with info from the Harpo's mock data. Based on our Figma prototype. Still very rough but this version is likely what's going to be in the midterm demo video.

ErikaJZhou committed:
remove address from POI bubble

jm564 commented on CityAssistant/CityAssistant/Views/CityARView.swift:
```diff
@@ -35,16 +35,46 @@ struct CameraView: UIViewRepresentable {
+//        let y = 0.5
+//        let rectOrigin = CGPoint(x: x, y: y)
+//        let rectSize = CGSize(width: 0, height: 0)
+//        let textFrameRect = CGRect(origin: rectOrigin, size: rectSize)
```
need to remove commented out code

jm564 commented on CityAssistant/CityAssistant/Views/CityARView.swift:
```diff
@@ -35,16 +35,46 @@ struct CameraView: UIViewRepresentable {
+        placePOI(poi: businessTypeText, arView: arView)
+        let addressText = create3dText(text: CameraView.mockPOI.Address.ToString(), x: -0.45, y: 0.2, fontSize: 0.07)
+        placePOI(poi: addressText, arView: arView)
+        let hours = "8 am to 9 pm"
```
Dont hardcode values, will make it more difficult to use this later

jm564 commented on CityAssistant/CityAssistant/Views/CityARView.swift:
```diff
@@ -35,16 +35,46 @@ struct CameraView: UIViewRepresentable {
+//        let myAnchor = AnchorEntity()
+//        myAnchor.addChild(myTxt)
+//        arView.scene.addAnchor(myAnchor)
+        let businessNameText = create3dText(text: CameraView.mockPOI.BusinessName, x: -0.45, y: 0.4)
```
This should all be in its own function something like create bubble (POI)

ErikaJZhou replied:
Moved to a new function createPOIBubble()

jm564 commented on CityAssistant/CityAssistant/Views/CityARView.swift:
```diff
@@ -54,6 +84,45 @@ struct CameraView: UIViewRepresentable {
+    // Create custom 3D text at specified local coordinate (x,y)
+    func create3dText(text: String, x: Double = 0.0, y: Double = 0.0, fontSize: CGFloat = 0.1) -> ModelEntity {
+        // Test placing text at location
+        let x = x
```
What is the point of x and y variables?

ErikaJZhou replied:
Default values in case they're not provided, but specifies where the text appears

ErikaJZhou commented:
![poi-bubble-no-address](https://github.com/Augmenters/FrontStone/assets/55419983/bf7c3bc4-f196-41bf-81bd-b30e03d55676)
Address was removed from the POI bubble.

Jreini committed:
Fixed bugs with AR View and and backstone changes

ErikaJZhou committed:
moved POI Bubble creation into createPOIBubble function

ErikaJZhou committed:
cleaned up createPOIBubble a little

ErikaJZhou committed:
Merge remote-tracking branch 'remotes/origin/refactor' into poi-bubble

ErikaJZhou committed:
brought my poi bubble code in. TODO merge with functions from Jules

ErikaJZhou committed:
Integrate POI placement and bubble creation

placePOI function now creates and places a POI bubble. Using newPosition for AnchorEntity breaks it

Co-authored by: Andrew Brain <andrewbrain2019@gmail.com>

ErikaJZhou committed:
amend previous commit - Integrate POI placement and bubble creation

placePOI function creates and places a POI bubble. Bug that bubble does not appear when newPosition is used for AnchorEntity

Co-authored by: Andrew Brain <andrewbrain2019@gmail.com>

ErikaJZhou committed:
amend previous commit again - Integrate POI placement and bubble creation

placePOI function creates and places a POI bubble. Bug that bubble does not appear when newPosition is used for AnchorEntity

Co-authored by: Andrew Brain <andrewbrain2019@gmail.com>

ErikaJZhou committed:
amend commit Integrate POI bubble creation and placement

Co-authored by: Andrew Brain <andrewbrain2019@gmail.com>

ErikaJZhou commented:
![IMG_532FFE8558CA-1](https://github.com/Augmenters/FrontStone/assets/55419983/eb609c92-a1aa-4999-bc44-71a7fd18d4cb)
@ABrain7710 and I worked on integrating @jm564 POI placement with bubble creation. Now the placePOI() function creates and places a POI. Using the newLocation to init the AnchorEntity caused the bubble to not show up.
Still has bug where bubble is not tappable.

ErikaJZhou commented:
The "refactor" branch was merged into this one.

Jreini committed:
Updated viewmodels to be observed models
